src/iotjs_module_httpparser.cpp

```diff
@@ -0,0 +1,566 @@
+/* Copyright 2015 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "iotjs_def.h"
+#include "iotjs_module_httpparser.h"
+#include "http_parser.h"
+#include "iotjs_objectwrap.h"
+#include "iotjs_module_buffer.h"
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+
+namespace iotjs {
+
+
+#define JSETPROPERTY( container_, fname_, vname_ )                      \
+  do {                                                                  \
+    JObject jobj(vname_ );                                              \
+    container_.SetProperty(fname_, jobj);                               \
+  } while(0)
+
+// increse this to minimize inter JS-C call
+#define HEADER_MAX 10
+
+
+struct StringHelper {
+  StringHelper() {
+    saved_heap = false;
+    Reset();
+  }
+
+  ~StringHelper() {
+    Reset();
+  }
+
+  void Reset() {
+    if (saved_heap) {
+      saved_heap = false;
+      delete str;
+    }
+    str = NULL;
+    len = 0;
+  }
+
+  void SaveToHeap() {
+    if (!saved_heap && len > 0) {
+      char* heap_str = new char[len];
+      memcpy(heap_str, str, len);
+      str = heap_str;
+      saved_heap = true;
+    }
+  }
+
+  void UpdateStr(const char* newstr, int length) {
+    if (str == NULL) {
+      str = newstr;
```

Comment:
```newstr``` is a part of local buffer being parsed and handled by http-parser.  It is safe since these strings are flushed to JS side before parsing is done. 
Anyway, I will replace StringHelper with String soon.